src/librustdoc/html/highlight.rs

```diff
@@ -254,23 +313,37 @@ impl<'a> Classifier<'a> {
                 .unwrap_or(false)
             {
                 let tokens = self.get_full_ident_path();
+                let skip = !tokens.is_empty();
                 for (token, start, end) in tokens {
```

Comment:
The skip variable is unnecessary; you never modify tokens so you can iterate over a reference instead and check is_empty() directly.

Reply:
Absolutely! Good catch.

Reply:
Actually no because `tokens` is consumed in the `for` loop just below. I added a comment so people don't fall in the same trap as us. :rofl:

Reply:
You don't have to consume tokens. You can iterate over &tokens instead.

Reply:
:open_mouth: How did I not see that?! Thanks!